typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -135,11 +138,22 @@ export abstract class HyperlaneDeployer<
       this.startingBlockNumbers[chain] = await this.multiProvider
         .getProvider(chain)
         .getBlockNumber();
-      await runWithTimeout(this.chainTimeoutMs, async () => {
+
+      const deployPromise = runWithTimeout(this.chainTimeoutMs, async () => {
         const contracts = await this.deployContracts(chain, configMap[chain]);
         this.addDeployedContracts(chain, contracts);
+        this.logger.info({ chain }, 'Successfully deployed contracts');
       });
+      if (this.options.concurrentDeploy) {
+        deployPromises.push(deployPromise);
+      } else {
+        await deployPromise;
+      }
     }
+
+    // Await all deploy promises. If concurrent deploy is not enabled, this will be a no-op.
+    await Promise.all(deployPromises);
```

Comment:
I would recommend using `Promise.allSettled` so 1 failure doesnt waste N-1 successes

Reply:
good shout, will do that